common/src/pri/defs.rs

```diff
@@ -23,6 +23,7 @@ pub trait ProgramRuntimeInterface {
     type AtomicBinaryOp;
     type DebugInfo;
     type Tag;
+    type MemoryOp;
```

Comment:
I think introducing an enum for this two-case entity is a bit overkill. Especially that we probably won't care much about that information. How about passing a boolean?